src/client/Microsoft.Identity.Client/Http/Retry/HttpRetryCondition.cs

```diff
@@ -62,6 +62,21 @@ public static bool RegionDiscovery(HttpResponse response, Exception exception)
             return (int)response.StatusCode is not (404 or 408);
         }
 
+        /// <summary>
+        /// Retry policy specific to CSR Metadata Probe.
+        /// Extends Imds retry policy but excludes 404 and 408 status codes.
+        /// </summary>
+        public static bool CsrMetadataProbe(HttpResponse response, Exception exception)
+        {
+            if (!Imds(response, exception))
+            {
+                return false;
+            }
+
+            // If Imds would retry but the status code is 404, don't retry
+            return (int)response.StatusCode is not 404;
```

Comment:
So, it will retry on all the failures except for 404? Or should it match the retry for all the status codes for IMDS and exclude the 404? https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/5359/files#diff-657c8db85ac31ecc939bedba01c7298d8b9f549926c0f52484b20e5a01cca01fR40